src/typechecker/FStar.TypeChecker.Rel.fs

```diff
@@ -2918,7 +2966,7 @@ and solve_t' (env:Env.env) (problem:tprob) (wl:worklist) : solution =
         let force_eta t =
             if is_abs t then t
             else begin
-                let _, ty, _ = env.type_of ({env with lax=true; use_bv_sorts=true; expected_typ=None}) t in
+                let _, ty, _ = env.typeof_tot_or_gtot_term ({env with lax=true; use_bv_sorts=true; expected_typ=None}) t true in  //AR: TODO: type_of_well_typed?
```

Comment:
Would be nice to use type_of_well_typed and remove the lax=true ... but that can be a separate PR